docs/adr/0002-docker-images-for-ruby-applications.md

```diff
@@ -0,0 +1,104 @@
+# 2. Docker images for Ruby Applications
+
+Date: 2020-11-17
+
+## Context
+
+In order to run GOV.UK applications in a Docker environment we need
+[Docker images][]. GOV.UK had previously created Docker images for various
+applications to enable the [publishing-e2e-tests][] suite, which runs a
+containerised version of the GOV.UK publishing stack. These images were
+defined with a Dockerfile in the root of a projects repository
+([example][example-dockerfile]) and, as they are the only Dockerfile for a
+project, were essentially the defacto Docker images for GOV.UK applications.
+
+Attempting to reuse these images in the GOV.UK Docker development environment
+revealed a number of problems:
+
+1. The images [embedded the applications files][embed-files] - this meant that
```

Comment:
This is relatively easy to overcome by mounting the local files over the top of those in the volume. We could say that, while there are workarounds for some of the issues here, on the whole using the existing images was impractical.

Reply:
Oh sure, I hadn't realised that - but seems pretty obvious now 🤦 . So rather than a total pain it's more of a redundant action that can get in a bit of a mess for files like Gemfile or package.json which impact container and runtime.

I've moved this concern from the top to the bottom of list to reflect it not being a deal breaker.